auth/schema/mutations/register.py

```diff
@@ -25,23 +26,43 @@ def validate_password(password):
         password_valid = password_length.compare(8, len(password))
         if password_valid == False:
             raise ValidationError("Password must be at least 8 characters")
+        # add regex for making sure password has characters and symbols
         return password
 
     def validate_email(email):
-        email_validator = EmailValidator()
-        email_valid = email_validator.__call__(email)
+        email_validator = EmailValidator(email)
         if email_valid == False:
             raise ValidationError()
         return email
-        
+
+    '''Validate a username and make sure it contains only alphanumeric characters and special characters'''
+    def validate_username(username):
+        username_validator = RegexValidator(
+            r"(^[~0-9A-Za-z]*)",
+            "Username must contain only alphanumeric characters and special characters",
+            None,
+            re.IGNORECASE)
+        valid_username = username_validator(username)
+        if valid_username == True:
+            raise ValidationError()
+        return username
+    
+    def validate_name(name):
+        if name.isalnum() == False:
+            raise ValidationError("Name must contain only alphanumeric characters")
+        return name
+
     def mutate(self, info, **kwargs):
-        first_name = kwargs.get('first_name')
-        last_name = kwargs.get('last_name')
+        raw_first_name = kwargs.get('first_name')
+        raw_last_name = kwargs.get('last_name')
         username = kwargs.get('username')
-        email = kwargs.get('email')
+        raw_email = kwargs.get('email')
         raw_password = kwargs.get('password')
         valid_password = RegisterUser.validate_password(raw_password)
-        email = RegisterUser.validate_email(email)
+        email = RegisterUser.validate_email(raw_email)
+        # username = RegisterUser.validate_username(raw_username)
```

Comment:
We should set up linting for our project to help catch these, to help keep our codebase clean

Reply:
Yes we have a story on the PT board for that I think

Reply:
however, that particular line is a placeholder reminder to add the username validations